### PR TITLE
fix issue #1361

### LINKED ISSUE #1361: [kube-up][scale-out]: code dead lock in function start-mizar-scaleout
**What happened**:
there are dead lock in function start-mizar-scaleout
```
function start-mizar-scaleout {
...
echo "Waiting until nodes RPs are ready"
until [ -f "/etc/srv/kubernetes/kube-scheduler/rp-kubeconfig-1" ]; do
   sleep 5
done
...
}
```
start-cluster-networking -> start-mizar -> start-mizar-scaleout running only on TP server. 
Scale-out starting process:
TP Servers -> RP servers -> proxy

when start mizar, RP servers are not  ready. this until will be dead lock. 

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
Added comments and found the logs in  /var/log/master-init.log
```
echo "Waiting until nodes RPs are ready"
  until [ -f "/etc/srv/kubernetes/kube-scheduler/rp-kubeconfig-1" ]; do
    echo "until dead lock"
    date
    sleep 5
  done
  date
  kubectl create configmap system-source --namespace=kube-system --from-literal=name=arktos --from-literal=company=futurewei

```

```
Deploying mizar daemonset
daemonset.apps/mizar-daemon created
Waiting until nodes RPs are ready
until dead lock
Tue Feb 15 03:20:49 UTC 2022
until dead lock
Tue Feb 15 03:20:54 UTC 2022
until dead lockTue Feb 15 03:20:59 UTC 2022
until dead lock
Tue Feb 15 03:21:04 UTC 2022
Deploying mizar daemonset
daemonset.apps/mizar-daemon created
Waiting until nodes RPs are ready
until dead lock
Tue Feb 15 03:20:49 UTC 2022
...
until dead lock
Tue Feb 15 03:27:45 UTC 2022
until dead lock
Tue Feb 15 03:27:50 UTC 2022
Tue Feb 15 03:27:55 UTC 2022
configmap/system-source created
```

Repro steps:
```
$ unset KUBE_GCE_MASTER_PROJECT KUBE_GCE_NODE_PROJECT KUBE_GCI_VERSION  KUBE_GCE_MASTER_IMAGE  KUBE_GCE_NODE_IMAGE KUBE_CONTAINER_RUNTIME NETWORK_PROVIDER
$ export KUBEMARK_NUM_NODES=100 NUM_NODES=1 SCALEOUT_CLUSTER=true SCALEOUT_TP_COUNT=1 SCALEOUT_RP_COUNT=1 RUN_PREFIX=sonyaperf2-021422
$ export MASTER_DISK_SIZE=500GB MASTER_ROOT_DISK_SIZE=500GB KUBE_GCE_ZONE=us-west2-b MASTER_SIZE=n1-highmem-32 NODE_SIZE=n1-highmem-16 NODE_DISK_SIZE=500GB GOPATH=$HOME/go KUBE_GCE_ENABLE_IP_ALIASES=true KUBE_GCE_PRIVATE_CLUSTER=true CREATE_CUSTOM_NETWORK=true KUBE_GCE_INSTANCE_PREFIX=${RUN_PREFIX} KUBE_GCE_NETWORK=${RUN_PREFIX} ENABLE_KCM_LEADER_ELECT=false ENABLE_SCHEDULER_LEADER_ELECT=false ETCD_QUOTA_BACKEND_BYTES=8589934592 SHARE_PARTITIONSERVER=false LOGROTATE_FILES_MAX_COUNT=200 LOGROTATE_MAX_SIZE=200M KUBE_ENABLE_APISERVER_INSECURE_PORT=true KUBE_ENABLE_PROMETHEUS_DEBUG=true KUBE_ENABLE_PPROF_DEBUG=true TEST_CLUSTER_LOG_LEVEL=--v=2 HOLLOW_KUBELET_TEST_LOG_LEVEL=--v=2 GCE_REGION=us-west2-b NETWORK_PROVIDER=mizar
$ ./cluster/kube-up.sh
```
**Anything else we need to know?**:

if ubuntu 18.04, script will stop on this loop.

**Environment**:
- Arktos version (use `kubectl version`):
https://github.com/CentaurusInfra/arktos/commit/2b6855fa7b204870620421cb1c533b182acbc812

- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by sonyafenge**
**What type of PR is this?**

> /kind bug

**What this PR does / why we need it**:
Fix issue #1361. Deploy Mizar-operator to all tps after all tp and rp server ready.
Only impact kube-up: scale-out

**Which issue(s) this PR fixes**:

Fixes #1361 

**Special notes for your reviewer**:
Test scale-out 1x1 and 2x2, both can get all mizar component start successfully.
```
$ kubectl --kubeconfig=/home/sonyali/go/src/k8s.io/arktos/cluster/kubeconfig.tp-1 get pods -A -o wide | grep mizar
default       mizar-daemon-fclm5                                        1563435164262111223   1/1     Running             0          18m   10.40.0.6   sonyaperf2-021522-rp-1-minion-group-5364   <none>           <none>
default       mizar-daemon-nf56m                                        7334669340671240868   1/1     Running             0          18m   10.40.0.5   sonyaperf2-021522-rp-2-master              <none>           <none>
default       mizar-daemon-nlngt                                        3206226579230392509   1/1     Running             0          18m   10.40.0.7   sonyaperf2-021522-rp-2-minion-group-k4r7   <none>           <none>
default       mizar-daemon-rbr5x                                        2713279784547345511   1/1     Running             0          18m   10.40.0.4   sonyaperf2-021522-rp-1-master              <none>           <none>
default       mizar-daemon-wkbl4                                        3413360866183995272   1/1     Running             0          35m   10.40.0.2   sonyaperf2-021522-tp-1-master              <none>           <none>
default       mizar-operator-5c759c674d-6qlxt                           1224469615237973248   1/1     Running             0          18m   10.40.0.2   sonyaperf2-021522-tp-1-master              <none>           <none>

$ kubectl --kubeconfig=/home/sonyali/go/src/k8s.io/arktos/cluster/kubeconfig.tp-2 get pods -A -o wide | grep mizar
default       mizar-operator-6659687458-r6hs9                           8200532644199407059   1/1     Running             0          19m   10.40.0.3   sonyaperf2-021522-tp-2-master              <none>           <none>
```
**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cluster/gce/gci/configure-helper-common.sh [hunk 1]
```diff
@@ -3257,6 +3257,7 @@ function wait-until-mizar-ready {
   done
 }
 
+# mizar-operator will be deployed after rp sesrver ready
```
Comment (Sindica) [inline cluster/gce/gci/configure-helper-common.sh:3260]: The issue is with waiting for RP ready not apply mizar operator. Prefer to have mizar component deployment in a single place.

Also can try call start-mizar-scaleout after all components started.

Comment (sonyafenge) [reply cluster/gce/gci/configure-helper-common.sh]: updated all mizar to start  after tp and rp ready

**Comment by sonyafenge** [comment]
Retested scale-out 1x1 and 2x2, both can get all mizar component start successfully.
```
$ kubectl --kubeconfig=/home/sonyali/go/src/k8s.io/arktos/cluster/kubeconfig.tp-1 get pods -A | grep mizar
default       mizar-daemon-bdssw                                        6637428862566857486   1/1     Running             0          40m
default       mizar-daemon-fc74w                                        2791265796256706550   1/1     Running             0          40m
default       mizar-daemon-jgmz5                                        6923580046602754646   1/1     Running             0          40m
default       mizar-daemon-wjlgn                                        2498704369955686196   1/1     Running             0          40m
default       mizar-operator-5c759c674d-4b9vv                           2098494063691682799   1/1     Running             0          40m
```

```
$ kubectl --kubeconfig=/home/sonyali/go/src/k8s.io/arktos/cluster/kubeconfig.tp-2 get pods -A | grep mizar
default       mizar-operator-6659687458-jrddp                           4467455810534808552   1/1     Running             0          40m
```

File: cluster/gce/util.sh [hunk 2]
```diff
@@ -3406,6 +3407,42 @@ function restart_tp_scheduler_and_controller {
...
+
+    if [[ "${tp_num}" == "1" ]]; then
+      # Place mizar daemon yaml.
+      cp "${src_dir}/mizar-daemon.yaml" "${dst_dir}"
+      src_file="${dst_dir}/mizar-daemon.yaml"
+      sed -i -e "s@{{network_provider_version}}@${NETWORK_PROVIDER_VERSION}@g" "${src_file}"
+      kubectl --kubeconfig=${tp_kubeconfig}  apply -f "${src_file}"
+    fi
+
+    kubectl --kubeconfig=${tp_kubeconfig} create configmap system-source --namespace=kube-system --from-literal=name=arktos --from-literal=company=futurewei
```
Comment (h-w-chen) [inline cluster/gce/util.sh:3431]: does it matter to move this line ahead of creating daemonset?

File: cluster/gce/gci/configure-helper-common.sh [hunk 3]
```diff
@@ -3309,9 +3271,8 @@ function start-mizar {
     sleep 5
   done
   kubectl label node `hostname` node-role.kubernetes.io/master=""
-  if [[ "${SCALEOUT_CLUSTER:-false}" == "true" ]]; then
-    start-mizar-scaleout
-  else
+  #For scaleout cluster, mizar will started after all tp and rp ready
```
Comment (h-w-chen) [inline cluster/gce/gci/configure-helper-common.sh:3274]: re-ordering the mizar installation seems good; are the checking readinesses of TP & RP able to identify the instance as ready w/o mizar installation initially?

**Review by h-w-chen — APPROVED** [review]
lgtm.
left a question to author; it is not blocking as it has been confirmed able to start successfully.

File: cluster/gce/gci/configure-helper-common.sh (hunk 3 above)
Comment (sonyafenge) [reply cluster/gce/gci/configure-helper-common.sh:3274]: yes, tp/rp identified the instance as ready even without chi plugin installed.

**Pull request merged**